lib/monitor.js

```diff
@@ -207,8 +169,9 @@ class Monitor {
 
         // Do a setImmediate here so that all the streams listening for "end" have a chance to run
         // https://github.com/nodejs/node/blob/master/lib/_stream_readable.js#L894-L897
-        setImmediate(callback);
+        return new Promise((resolve) => setImmediate(() => resolve()));
```

Comment:
Curious if we still need this?

If so you can write it as `return new Promise((resolve) => setImmediate(resolve));`

Reply:
I don't think so so removing